src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/UniqueAccessCodeServiceImpl.java

```diff
@@ -44,88 +43,70 @@
 
   private static final Logger log = LoggerFactory.getLogger(UniqueAccessCodeService.class);
 
+  @Autowired private AppConfig appConfig;
   @Autowired private RespondentDataRepository dataRepo;
   @Autowired private EventPublisher eventPublisher;
-
-  private BoundMapperFacade<UAC, UniqueAccessCodeDTO> uacMapperFacade;
-  private BoundMapperFacade<CollectionCase, UniqueAccessCodeDTO> caseMapperFacade;
+  @Autowired private MapperFacade mapperFacade;
 
   // Enums to capture the linking matrix of valid form type and case types.
   // Original table is from:
   // https://collaborate2.ons.gov.uk/confluence/display/SDC/Auth.05+-+Unlinked+Authentication
   private enum LinkingCombination {
-    H1(FormType.H, CaseType.HH, CaseType.HH),
-    H2(FormType.H, CaseType.HH, CaseType.SPG),
-    H3(FormType.H, CaseType.HH, CaseType.CE),
-    H4(FormType.H, CaseType.SPG, CaseType.HH),
-    H5(FormType.H, CaseType.SPG, CaseType.SPG),
-    H6(FormType.H, CaseType.SPG, CaseType.CE),
-    I1(FormType.I, CaseType.HH, CaseType.HH),
-    I2(FormType.I, CaseType.HH, CaseType.SPG),
-    I3(FormType.I, CaseType.HH, CaseType.CE),
-    I4(FormType.I, CaseType.SPG, CaseType.HH),
-    I5(FormType.I, CaseType.SPG, CaseType.SPG),
-    I6(FormType.I, CaseType.SPG, CaseType.CE),
-    I7(FormType.I, CaseType.CE, CaseType.HH),
-    I8(FormType.I, CaseType.CE, CaseType.SPG),
-    I9(FormType.I, CaseType.CE, CaseType.CE),
-    C1(FormType.CE1, CaseType.CE, CaseType.CE);
+    H1(FormType.H, CaseType.HH),
+    H2(FormType.H, CaseType.SPG),
+    H3(FormType.H, CaseType.CE),
+    I1(FormType.I, CaseType.HH),
+    I2(FormType.I, CaseType.SPG),
+    I3(FormType.I, CaseType.CE),
+    C1(FormType.CE1, CaseType.CE);
```

Comment:
We have discussed the value of keeping this much reduced table. I think there is a pretty good case for keeping it. If this information is not kept a table it'll end up as a bunch of if statements.
Not sure about the enum names though (my fault !!).